pkg/common/resources/quantity.go

```diff
@@ -0,0 +1,103 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package resources
+
+import (
+	"errors"
+	"math/big"
+	"regexp"
+	"strconv"
+)
+
+// This code handles parsing of SI units in quantities:
+// <quantity>     ::= <signedNumber><suffix>
+// <signedNumber> ::= <digits> | <sign><digits>
+// <sign>         ::= "+" | "-"
```

Comment:
In the config(s) we do not allow negative values. I don't think we should allow that here.

Reply:
Removed negative support.